src/Adapter/MSTest.TestAdapter/Helpers/AttributeComparer.cs

```diff
@@ -6,7 +6,7 @@ namespace Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Helpers;
 internal static class AttributeComparer
 {
     public static bool IsNonDerived<TAttribute>(Attribute attribute) =>
-        attribute is TAttribute;
+        attribute.GetType() == typeof(TAttribute);
```

Comment:
do you know if compiler will optimize this call to be about the same or better perf? I don't think it will create the whole type object and compare it. But I am not 100% sure :)

Reply:
`GetType() == typeof(...)` is generally faster than `is` type checking as far as I know. The compiler doesn't need to optimize actually

Reply:
I have learnt that earlier from https://github.com/dotnet/runtime/blob/1d86001220b04cd9d074c28ab62610c6acb7698e/src/libraries/System.Linq/src/System/Linq/Enumerable.cs#L42-L47 :)